sqlglot/dialects/dialect.py

```diff
@@ -135,6 +144,11 @@ def get_start_end(token_type: TokenType) -> t.Tuple[t.Optional[str], t.Optional[
 
         klass.generator_class.can_identify = klass.can_identify
 
+        if klass.TYPED_DIVISION:
+            klass.parser_class.FACTOR = {**klass.parser_class.FACTOR, TokenType.SLASH: exp.TypedDiv}
```

Comment:
i'm not sure how easy it is to do this for the parser because it's a constant

Reply:
yeah I think that's why we also have this segment right below it:

```python
        if not klass.STRICT_STRING_CONCAT and klass.DPIPE_IS_STRING_CONCAT:
            klass.parser_class.BITWISE[TokenType.DPIPE] = exp.SafeDPipe
```

Reply:
So we add the TYPED_DIVISION attribute to the generator, but not the parser?
And then mutate the parser here in `dialect`, but conditionally change the generation logic in the _sql methods?

Reply:
yea, i think that's how we currently do things